src/components/Header/index.jsx

```diff
@@ -30,16 +31,35 @@ class HeaderComponent extends React.PureComponent<Props, State> {
     this.onConnect()
   }
 
+  componentDidCatch(error: Error, info: Info) {
+    const { openSnackbar } = this.props
+    this.setState({ hasError: true })
+    openSnackbar(WALLET_ERROR_MSG, 'error')
+
+    logComponentStack(error, info)
+  }
+
   onDisconnect = () => {
     const { removeProvider, openSnackbar } = this.props
+    clearInterval(this.providerListener)
 
     removeProvider(openSnackbar)
   }
 
-  onConnect = () => {
+  onConnect = async () => {
     const { fetchProvider, openSnackbar } = this.props
 
-    fetchProvider(openSnackbar)
+    clearInterval(this.providerListener)
+    let currentProvider: ProviderProps = await getProviderInfo()
+    fetchProvider(currentProvider, openSnackbar)
```

Comment:
Just one thing, if fetching provider is done in `getProviderInfo` now, I think it makes sense to rename the action to `setProvider` or something

Reply:
Yes, thanks!